### DISCUSSION TITLE
How can I know when the tokens are finished?

### DISCUSSION THREAD

**Discussion by pironev** [withcatai/node-llama-cpp#370] [Q&A] [answered]
I created a stream on the server like this:
```
const readable = new Readable({read() {}});
session.prompt("How to call a function from a module in Python?", {
  onTextChunk(chunk) {
    readable.push(chunk);
  }
});
return reply.send(readable);
```
But for the client I need to close the stream. So I need to execute readable.push(null);
Is there an event to know when is finished the loop from onTextChunk?
Thank you!

**Comment by giladgd** [comment, marked as answer]
You need to `await` on the promise returned from `session.prompt(...)` to know when the generation has finished.
